types/modules.d.ts

```diff
@@ -1,6 +1,64 @@
 // TODO: create types for these modules
 declare module 'hyperswarm'
-declare module '@hyperswarm/secret-stream'
+declare module '@hyperswarm/secret-stream' {
+  import { Duplex as NodeDuplex } from 'stream'
+  import { Duplex, DuplexEvents } from 'streamx'
+
+  interface Opts {
+    autostart?: boolean
+    // TODO: Use https://github.com/chm-diederichs/noise-handshake/blob/main/noise.js for specific patterns
+    pattern?: string
```

Comment:
tbh i had no idea how to read those pattern variables in the link provided, so went with string for now

Reply:
yeah i think we're unlikely to set the pattern option too, so seems lower priority.